master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -1036,6 +1036,7 @@ private[celeborn] class Master(
     nonEagerHandler.submit(new Runnable {
       override def run(): Unit = {
         workersAssignedToApp.remove(appId)
+        secretRegistry.unregister(appId);
```

Comment:
Discuss: I have a mild concern about the split logic to cleanup here ... meta cleanup happens in `statusSystem.handleAppLost` vs unregisteration from secret registery here.